File: obsolete/src/tauri-adapter/stats.ts
import { invoke } from "./invoke";

export interface TrendDataPoint {
  date: string;
  income: number;
  expense: number;
}

export interface CategoryBreakdown {
  category_name: string;
  amount: number;
  percent: number;
}

export interface AssetData {
  account_type: string;
  balance: number;
}

export interface ComparisonData {
  current_income: number;
  current_expense: number;
  previous_income: number;
  previous_expense: number;
  income_change_percent: number;
  expense_change_percent: number;
}

export interface TagStat {
  tag: string;
  income: number;
  expense: number;
}

export interface RankedTransaction {
  id: string;
  amount: number;
  merchant: string | null;
  notes: string | null;
  transaction_date: number;
  transaction_type: "expense" | "income";
  category_name: string | null;
}

export interface DashboardSummary {
  income: number;
  expense: number;
  net_cash_flow: number;
  net_assets: number;
  transaction_count: number;
}

export const StatsAPI = {
  getMonthlyTrend: async (ledgerId: string, startTs: number, endTs: number): Promise<TrendDataPoint[]> => {
    return invoke("get_monthly_trend", { ledgerId, startTs, endTs });
  },

  getTrend: async (ledgerId: string, startTs: number, endTs: number, granularity: "day" | "week" | "month" | "year"): Promise<TrendDataPoint[]> => {
    return invoke("get_trend", { ledgerId, startTs, endTs, granularity });
  },
  
  getCategoryBreakdown: async (ledgerId: string, startTs: number, endTs: number, txType: "expense" | "income"): Promise<CategoryBreakdown[]> => {
    return invoke("get_category_breakdown", { ledgerId, startTs, endTs, txType });
  },
  
  getAssetsOverview: async (): Promise<AssetData[]> => {
    return invoke("get_assets_overview");
  },

  getComparison: async (ledgerId: string, currentStart: number, currentEnd: number, previousStart: number, previousEnd: number): Promise<ComparisonData> => {
    return invoke("get_comparison", { ledgerId, currentStart, currentEnd, previousStart, previousEnd });
  },

  getTagStats: async (ledgerId: string, startTs: number, endTs: number): Promise<TagStat[]> => {
    return invoke("get_tag_stats", { ledgerId, startTs, endTs });
  },

  getTopTransactions: async (ledgerId: string, startTs: number, endTs: number, txType: "expense" | "income", limit: number): Promise<RankedTransaction[]> => {
    return invoke("get_top_transactions", { ledgerId, startTs, endTs, txType, limit });
  },

  getDashboardSummary: async (ledgerId: string, startTs: number, endTs: number): Promise<DashboardSummary> => {
    return invoke("get_dashboard_summary", { ledgerId, startTs, endTs });
  }
};

File: obsolete/src-tauri/src/commands/stats.rs
use crate::adapters::sqlite_store::SqliteStore;
use crate::core::stats_engine::{
    AssetData, CategoryBreakdown, ComparisonData, DashboardSummary, RankedTransaction, TagStat,
    TrendDataPoint,
};
use crate::error::AppError;
use tauri::State;

#[tauri::command]
pub fn get_monthly_trend(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    start_ts: i64,
    end_ts: i64,
) -> Result<Vec<TrendDataPoint>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_monthly_trend(&ledger_id, start_ts, end_ts)
}

#[tauri::command]
pub fn get_trend(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    start_ts: i64,
    end_ts: i64,
    granularity: String,
) -> Result<Vec<TrendDataPoint>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_trend(&ledger_id, start_ts, end_ts, &granularity)
}

#[tauri::command]
pub fn get_category_breakdown(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    start_ts: i64,
    end_ts: i64,
    tx_type: String,
) -> Result<Vec<CategoryBreakdown>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_category_breakdown(&ledger_id, start_ts, end_ts, &tx_type)
}

#[tauri::command]
pub fn get_assets_overview(store: State<'_, SqliteStore>) -> Result<Vec<AssetData>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_assets_overview()
}

#[tauri::command]
pub fn get_comparison(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    current_start: i64,
    current_end: i64,
    previous_start: i64,
    previous_end: i64,
) -> Result<ComparisonData, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_comparison(
        &ledger_id,
        current_start,
        current_end,
        previous_start,
        previous_end,
    )
}

#[tauri::command]
pub fn get_tag_stats(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    start_ts: i64,
    end_ts: i64,
) -> Result<Vec<TagStat>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_tag_stats(&ledger_id, start_ts, end_ts)
}

#[tauri::command]
pub fn get_top_transactions(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    start_ts: i64,
    end_ts: i64,
    tx_type: String,
    limit: i64,
) -> Result<Vec<RankedTransaction>, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_top_transactions(&ledger_id, start_ts, end_ts, &tx_type, limit)
}

#[tauri::command]
pub fn get_dashboard_summary(
    store: State<'_, SqliteStore>,
    ledger_id: String,
    start_ts: i64,
    end_ts: i64,
) -> Result<DashboardSummary, AppError> {
    let engine = crate::core::stats_engine::StatsEngine::new(store.get_conn());
    engine.get_dashboard_summary(&ledger_id, start_ts, end_ts)
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn stats_command_surface_exposes_analysis_apis() {
        let _ = get_trend;
        let _ = get_comparison;
        let _ = get_tag_stats;
        let _ = get_top_transactions;
        let _ = get_dashboard_summary;
    }
}
